src/turn-command-manager.ts

```diff
@@ -616,3 +536,61 @@ export class TurnCommandManager {
     this.appliedMoveHeaders = true;
   }
 }
+
+// #region Comparators
+
+/**
+ * The main comparator used to sort turn commands. Consists of two sub-comparators
+ * that are sequentially applied:
+ * 1. Pre-Speed comparison modifiers, including command type, move priority, etc.
+ * 2. Speed comparison between the commands' source Pokemon.
+ *
+ * @see {@linkcode comparePreSpeed}
+ * @see {@linkcode compareSpeed}
+ */
+function compareTurnOrder(commandA: TurnCommand, commandB: TurnCommand): number {
+  return comparePreSpeed(commandA, commandB) || compareSpeed(commandA, commandB);
+}
+
+/**
+ * Prioritizes commands based on effects unrelated to the source Pokemon's Speed,
+ * including command type, move priority, and Speed-bypassing turn order modifiers
+ * such as the effects of Quash and Quick Draw.
+ */
+function comparePreSpeed(commandA: TurnCommand, commandB: TurnCommand): number {
+  if (commandA.command !== commandB.command) {
+    return COMMAND_PRIORITY_MAP[commandA.command] > COMMAND_PRIORITY_MAP[commandB.command] ? -1 : 1;
+  }
+  if (commandA.command === BattleCommand.FIGHT) {
+    const [aQuashed, bQuashed] = [commandA, commandB].map(({ pokemon }) => pokemon.hasTag(BattlerTagType.QUASHED));
+    if ((aQuashed || bQuashed) && aQuashed !== bQuashed) {
```

Comment:
```suggestion
    if (aQuashed !== bQuashed) {
```
If `aQuashed !== bQuashed` is `true`, then one or the other of them must be `true`, so the first half of the `if` is redundant.